### PR TITLE
feat(iam): an example for GenerateAccessToken

### LINKED ISSUE #6153: Document how grpc::AccessTokenCredentials works
To use these in the GUAC implementation we need to understand:

- [x] What exactly is the format for `access_token`?
- [x] Do these credentials automatically refresh or do we have to take action after some time?  In other words:
    - [x] Are new tokens created by gRPC? **Answer: no**
    - [x] Do we need to create a new `grpc::CallCredentials` object? **Answer: yes**

### PR DESCRIPTION
**Pull request by coryan**
I must confess that this is motivated to understand how the gRPC
credential types work. I think the example is useful, as it shows how to
call the API and (more interesting) how to use the results.

Fixes #6153

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6188)

### CODE REVIEW

File: google/cloud/iam/samples/grpc_credential_types.cc [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        grpc::AccessTokenCredentials(token.access_token()));
+    spanner::InstanceAdminClient admin(spanner::MakeInstanceAdminConnection(
+        google::cloud::Options{}.set<google::cloud::GrpcCredentialOption>(
+            credentials)));
+    for (auto instance : admin.ListInstances(project_id, /*filter=*/{})) {
+      // kUnauthenticated receives special treatment, it is the error received
+      // when the token expires.
+      if (instance.status().code() ==
+          google::cloud::StatusCode::kUnauthenticated) {
+        std::cout << "error [" << instance.status().code() << "]";
```
Comment (devbww) [inline google/cloud/iam/samples/grpc_credential_types.cc:93]: Just stream `instance.status()` perhaps?

File: google/cloud/iam/samples/grpc_credential_types.cc [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/iam/iam_credentials_client.gcpcxx.pb.h"
+#include "google/cloud/spanner/instance_admin_client.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/log.h"
+#include "google/cloud/testing_util/example_driver.h"
+#include "absl/strings/str_split.h"
+#include "absl/time/time.h"  // NOLINT
```
Comment (devbww) [inline google/cloud/iam/samples/grpc_credential_types.cc:21]: Q.  Do you recall what required the `NOLINT`?

**Review by devbww — APPROVED** [review]

File: google/cloud/iam/samples/grpc_credential_types.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/iam/samples/grpc_credential_types.cc]: Done.

File: google/cloud/iam/samples/grpc_credential_types.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/iam/samples/grpc_credential_types.cc]: I think it is this:

https://clang.llvm.org/extra/clang-tidy/checks/modernize-deprecated-headers.html

My guess is that clang-tidy is wrong here (maybe a regular expression that is accepting too much?). Separately, maybe my IDE settings are more strict that necessary, i.e., our builds may not yet care about this particular clang-tidy warning.

Despite all that, I am inclined to leave the NOLINT.

**Pull request merged**